core/src/main/scala/pickling/Macros.scala

```diff
@@ -31,6 +31,25 @@ trait TypeAnalysis extends Macro {
   }
 }
 
+// purpose of this macro: a wrapper around both pickler and unpickler
+trait PicklerUnpicklerMacros extends Macro {
+  import c.universe._
+
+  def impl[T: c.WeakTypeTag]: c.Tree = preferringAlternativeImplicits {
+    val originalTpe = weakTypeOf[T]
+    // Note: this makes it so modules work, things like foo.type.
+    //       For some reason we get an issue with not having == defined on Class[_] otherwise.
+    val tpe = {
+      // TODO - fix this for certain primitive types, like Null, etc.
```

Comment:
Would be nice to figure that out before we copy-paste.  at a minimum, we should have a re-usable method we can call so if we ever figure out WHY we widen, we can fix it in one spot.